src/Scripting/CSharp/Hosting/CommandLine/Csi.cs

```diff
@@ -18,15 +18,21 @@ internal CSharpInteractiveCompiler(string responseFile, string baseDirectory, st
         {
         }
 
+        internal override Type Type => typeof(CSharpInteractiveCompiler);
+
+        internal override string GetAssemblyFileVersion()
+        {
+            return Type.GetTypeInfo().Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version;
```

Comment:
I don't know, but I suspect that the difference is due to the compiler server (for the command-line compilers), where the compiler server loads the assembly itself from disk. The base implementation models that (i.e. tells you the version of the compiler that the compiler server will run).